### ISSUE TITLE
Publish app on f-droid

### ISSUE THREAD

**Issue by joe733** [jld3103/nextcloud-harbour#18]
[F-Droid | App Developers | FAQ](https://f-droid.org/docs/FAQ_-_App_Developers/)

**Comment by jld3103** [comment]
Template for submission later:
```
Categories:
  - Internet
License: BSD-3-Clause
AuthorName: jld3103
SourceCode: https://github.com/jld3103/nextcloud-harbour
Liberapay: jld3103

AutoName: Nextcloud Harbour

RepoType: git
Repo: https://github.com/jld3103/nextcloud-harbour

Builds:
  - versionName: 1.0.0
    versionCode: 1
    commit: dce04ed
    subdir: packages/harbour
    output: build/app/outputs/flutter-apk/app-release-unsigned.apk
    srclibs:
      - flutter@3.0.5
    build:
      - $$flutter$$/bin/flutter config --no-analytics
      - $$flutter$$/bin/flutter packages pub get
      - $$flutter$$/bin/flutter build apk --no-tree-shake-icons

AutoUpdateMode: Version
UpdateCheckMode: Tags
UpdateCheckData: packages/harbour/pubspec.yaml|version:\s.+\+(\d+)|.|version:\s(.+)\+
CurrentVersion: 1.0.0
CurrentVersionCode: 1
```

I think once I make a first proper `1.0.0` release I'll make a MR for inclusion in the official F-Droid repo.

**Comment by jld3103** [comment]
A nightly repository is now available at https://github.com/jld3103/nextcloud-neon-nightly !

**Comment by jld3103** [comment]
@joe733 Can you try it out?

**Comment by jld3103** [comment]
F-Droig nightly actually works now

**Comment by p1gp1g** [comment]
Steps to get a reproducible build on fdroid with flutter are listed here: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/12398

TLDR;
- flutter as a submodule of the git project
- Known build path for instance /tmp/build (`git clone https://foo /tmp/build`)

**Comment by wrenix** [comment]
The nightlt shows me a 404

**Comment by provokateurin** [comment]
@wrenix https://github.com/provokateurin/nextcloud-neon-nightly/ this?

**Comment by wrenix** [comment]
merci

**Comment by provokateurin** [comment]
Which link did you use before? I can fix it if you tell me where the wrong link is.

**Comment by alexandru0-dev** [comment]
@provokateurin did the repo got deleted? now it's empty

**Comment by provokateurin** [comment]
Yeah it's broken right now :/ I'll try to fix it soon

**Comment by provokateurin** [comment]
The nightly fixed itself somehow, I have no clue why

**Comment by provokateurin** [comment]
I know :grimacing:
There isn't really anything I can do about it. At some point the f-droid action just fails every time.
Usually clearing the entire repository works, but I did that already and it still doesn't work.

**Comment by provokateurin** [comment]
Thank you @wrenix!
The fix worked (https://github.com/nextcloud/neon/actions/runs/10546640045/job/29218364673?pr=2422), but neither the official F-Droid client nor NeoStore show the app to me.
The APKs are definitely there (https://github.com/provokateurin/nextcloud-neon-nightly/tree/master/fdroid/repo), but something else must still be going wrong :thinking:

**Comment by provokateurin** [comment]
Oh I was wrong, only the signatures are in there but the APKs are missing. Then it makes sense that I don't get the app displayed...

**Comment by provokateurin** [comment]
To add some further information:
The workflow stopped working on 12th of June.
There were two merges around that time:
https://gitlab.com/fdroid/fdroidserver/-/commit/aeb8a7a3e5cfe4e2360350167e67f2288cec8ce4
https://gitlab.com/fdroid/fdroidserver/-/commit/21e03e62a86775370ee58754f6264c02472b3bb9
Looking at the changes of the later, those could be related to the failure.

**Comment by provokateurin** [comment]
No luck with the previous commit or version from the ppa either :cry:

**Comment by provokateurin** [comment]
Nightly workflow will work properly again thanks to the debugging and fix by @wrenix !

**Comment by provokateurin** [comment]
It was automatically generated in the past, but that doesn't seem to happen anymore :thinking:
